lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -260,23 +260,70 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		log.Infof("Performing funding tx coin selection using %v "+
 			"sat/kw as fee rate", int64(r.FeeRate))
 
+		var (
+			// allCoins referres to the entirety of coins in our
+			// wallet that are available for funding a channel.
+			allCoins []Coin
+			// manuallySelectedCoins referres to the client-side
+			// selected coins that should be considered available
+			// for funding a channel.
+			manuallySelectedCoins []Coin
+			err                   error
+		)
+
+		// outpointsToCoins maps outpoints to coins in our wallet iff
+		// these coins are existent and returns an error otherwise.
+		outpointsToCoins := func(outpoints []*wire.OutPoint) ([]Coin,
+			error) {
+
+			var selectedCoins []Coin
```

Comment:
nit: since we know how many coins there are, would it be preferable to call `make([]Coin, len(outpoints))` and then write to each index within the for loop below?

Reply:
Thanks for the suggestion. I am unclear if there's a benefit to using indexes here. If so I would change it, but for readability I prefer the current code.